lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -251,24 +253,29 @@ private static function get_inner_blocks( $attributes, $block ) {
 		}
 
 		// Load inner blocks from the navigation post.
-		if ( array_key_exists( 'ref', $attributes ) ) {
+		if ( $navigation_post_id ) {
 			$inner_blocks = static::get_inner_blocks_from_navigation_post( $attributes );
 		}
 
 		// If there are no inner blocks then fallback to rendering an appropriate fallback.
 		if ( empty( $inner_blocks ) ) {
-			$inner_blocks = static::get_inner_blocks_from_fallback( $attributes );
+			$inner_blocks            = static::get_inner_blocks_from_fallback( $attributes );
+			$fallback_naviation_post = gutenberg_block_core_navigation_get_fallback_post();
+
+			if ( $fallback_naviation_post ) {
+				$navigation_post_id = $fallback_naviation_post->ID;
+			}
 		}
 
 		/**
 		 * Filter navigation block $inner_blocks.
 		 * Allows modification of a navigation block menu items.
 		 *
 		 * @since 6.1.0
-		 *
+		 * @param int|null $navigation_post_id The navigation post ID.
```

Comment:
This should go below the next line (so it corresponds to the filter's argument order) 🙂

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/57754/commits/f8d884486f3d3d425c6b88d0f276c7569083eaf7